### PR TITLE
C++: expose special key codes constants

### PR DESCRIPTION
**Pull request by ogoffart**

### CODE REVIEW

File: api/cpp/cbindgen.rs [hunk 1]
```diff
@@ -44,8 +44,36 @@ fn enums(path: &Path) -> anyhow::Result<()> {
...
     }
     i_slint_common::for_each_enums!(print_enums);
+
     writeln!(enums_pub, "}}")?;
     writeln!(enums_priv, "}}")?;
+
+    // Print the key codes constants
+    // This is not an enum, but fits well in that file
+    writeln!(enums_pub, r#"
+/// This namespace contains constants for each special non-printable keys.
```
Comment (tronical) [inline api/cpp/cbindgen.rs:54]:
```suggestion
/// This namespace contains constants for each special non-printable key.
```

File: api/cpp/cbindgen.rs [hunk 2]
```diff
@@ -44,8 +44,36 @@ fn enums(path: &Path) -> anyhow::Result<()> {
...
+///
+/// The constant can be converted to SharedString
```
Comment (tronical) [inline api/cpp/cbindgen.rs:56]:
```suggestion
/// Each constant can be converted to SharedString.
```

File: api/cpp/cbindgen.rs [hunk 3]
```diff
@@ -44,8 +44,36 @@ fn enums(path: &Path) -> anyhow::Result<()> {
...
+/// They are meant to be used with the slint::Window::dispatch_key_press_event() and
+/// slint::Window::dispatch_key_release_event().
```
Comment (tronical) [inline api/cpp/cbindgen.rs:58]:
```suggestion
/// The constants are meant to be used with the slint::Window::dispatch_key_press_event() and
/// slint::Window::dispatch_key_release_event() functions.
```

File: api/cpp/cbindgen.rs [hunk 4]
```diff
@@ -44,8 +44,36 @@ fn enums(path: &Path) -> anyhow::Result<()> {
...
+/// Example:
+/// ```
+/// window.dispatch_key_press_event(slint::platform::key_codes::Tab);
+/// ```
+namespace slint::platform::key_codes {{
+"#)?;
+    macro_rules! print_key_codes {
+        ($($char:literal # $name:ident # $($qt:ident)|* # $($winit:ident)|* ;)*) => {
+            $(
+                writeln!(enums_pub, "///A constant that represent the key code to be used in slint::Window::dispatch_key_press_event()")?;
```
Comment (tronical) [inline api/cpp/cbindgen.rs:69]:
```suggestion
                writeln!(enums_pub, "/// A constant that represents the key code to be used in slint::Window::dispatch_key_press_event()")?;
```

**Review by tronical — APPROVED** [review]
Worth a changelog entry, too :)

**Comment by ogoffart** [comment]
> Worth a changelog entry, too :)

It is already there in the "Added Platform API" line

**Pull request merged**